src/main/java/com/travelport/projectone/service/impl/ClientServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.travelport.projectone.service.impl;
+
+import com.travelport.projectone.entity.Client;
+import com.travelport.projectone.jpa.ClientJpaRepository;
+import com.travelport.projectone.service.ClientService;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ClientServiceImpl implements ClientService {
+
+    private final ClientJpaRepository clientDao;
+
+    public ClientServiceImpl(ClientJpaRepository clientDao) {
+        this.clientDao = clientDao;
+    }
+
+    @Override
+    public Optional<Client> save(Client client) {
+        if (client == null) return Optional.empty();
+        if (clientDao.existsById(client.getNif())) return Optional.empty();
+
+        try {
+            return Optional.of(clientDao.save(client));
+        } catch (Exception e) {
+            return Optional.empty();
```

Comment:
Es buena idea capturar errores, pero estos deben mostrarse siempre, de otra manera no se sabrá si hubo o no errores en el programa:
https://en.wikipedia.org/wiki/Error_hiding#:~:text=In%20computer%20programming%2C%20error%20hiding,anti%2Dpattern%20in%20computer%20programming.